pcdsdevices/sample_delivery.py

```diff
@@ -0,0 +1,435 @@
+        Estimated depletion time of 5.
+
+    start6_prefix : str
+        Starting volume of 6.
+
+    used6_prefix : str
+        Flow of 6.
+
+    time6_prefix : str
+        Estimated depletion time of 6.
+
+    start7_prefix : str
+        Starting volume of 7.
+
+    used7_prefix : str
+        Flow of 7.
+
+    time7_prefix : str
+        Estimated depletion time of 7.
+
+    start8_prefix : str
+        Starting volume of 8.
+
+    used8_prefix : str
+        Flow of 8.
+
+    time8_prefix : str
+        Estimated depletion time of 8.
+
+    start9_prefix : str
+        Starting volume of 9.
+
+    used9_prefix : str
+        Flow of 9.
+
+    time9_prefix : str
+        Estimated depletion time of 9.
+
+    start10_prefix : str
+        Starting volume of 10.
+
+    used10_prefix : str
+        Flow of 10.
+
+    time10_prefix : str
+        Estimated depletion time of 10.
+    """
+
+    integrator_source = UCpt(EpicsSignal)
+    flow_source = UCpt(EpicsSignal)
+    names = UCpt(EpicsSignal)
+
+    start1 = UCpt(EpicsSignal)
+    used1 = UCpt(EpicsSignal)
+    time1 = UCpt(EpicsSignal)
+
+    start2 = UCpt(EpicsSignal)
+    used2 = UCpt(EpicsSignal)
+    time2 = UCpt(EpicsSignal)
+
+    start3 = UCpt(EpicsSignal)
+    used3 = UCpt(EpicsSignal)
+    time3 = UCpt(EpicsSignal)
+
+    start4 = UCpt(EpicsSignal)
+    used4 = UCpt(EpicsSignal)
+    time4 = UCpt(EpicsSignal)
+
+    start5 = UCpt(EpicsSignal)
+    used5 = UCpt(EpicsSignal)
+    time5 = UCpt(EpicsSignal)
+
+    start6 = UCpt(EpicsSignal)
+    used6 = UCpt(EpicsSignal)
+    time6 = UCpt(EpicsSignal)
+
+    start7 = UCpt(EpicsSignal)
+    used7 = UCpt(EpicsSignal)
+    time7 = UCpt(EpicsSignal)
+
+    start8 = UCpt(EpicsSignal)
+    used8 = UCpt(EpicsSignal)
+    time8 = UCpt(EpicsSignal)
+
+    start9 = UCpt(EpicsSignal)
+    used9 = UCpt(EpicsSignal)
+    time9 = UCpt(EpicsSignal)
+
+    start10 = UCpt(EpicsSignal)
+    used10 = UCpt(EpicsSignal)
+    time10 = UCpt(EpicsSignal)
+
+    def __init__(self, prefix, *, name, **kwargs):
+        UCpt.collect_prefixes(self, kwargs)
+        super().__init__(prefix, name=name, **kwargs)
+
+
+class ManifoldValve(Device):
+    """
+    A single valve as present in the SDS Gas Manifold.
+
+    Parameters
+    ----------
+    prefix : str
+        The base PV for the Gas Manifold.
+
+    name : str
+        A name for the device.
+    """
+
+    open = Cpt(PytmcSignal, ':Open', io='o', kind='normal')
+    open_do = Cpt(PytmcSignal, ':OpenDO', io='i', kind='normal')
+    open_sw = Cpt(PytmcSignal, ':OpenSW', io='i', kind='normal')
+    interlocked = Cpt(PytmcSignal, ':Ilk', io='i', kind='normal')
+
+
+class GasManifold(Device):
+    """
+    A Gas Manifold as used in the sample delivery system.
+
+    Parameters
+    ----------
+    prefix : str
+        The base PV for the Gas Manifold.
+
+    name : str
+        A name for the device.
+    """
+
+    status = Cpt(PytmcSignal, ':IO:SyncUnitOK', io='i', kind='normal')
```

Comment:
Perhaps we should take a hint from Alex's TwinCAT implementation of this and also make a SDS base device that includes this signal so that we don't have to add it into all of these separately

Reply:
True